canonical-websites/docs.jujucharms.com
# Pretty URLs

This makes use of the new changes in `v0.2.0` of [canonicalwebteam.discourse-docs](https://github.com/canonical-web-and-design/canonicalwebteam.discourse-docs), to bring crafted URLs to documentation pages. So the changes are:

- URLs in the left navigation have been replaced with their "pretty" versions
- If you visit a topic URL, e.g. `/t/clouds/1100`, you should end up at the pretty URL, e.g. `/clouds`
- Topics that don't have pretty URLs defined will now be shown at `/t/{slug}/{id}` rather than `/{slug}/{id}`, because the solution to making this URL prettier should always be to create a proper URL mapping

QA
--

`./run`

- Check the homepage still appears as expected at http://127.0.0.1:8033
- Check pages redirect to pretty URLs - http://127.0.0.1:8033/t/clouds/1100, http://127.0.0.1:8033/clouds/1100, http://127.0.0.1:8033/t/1100, http://127.0.0.1:8033/1100 all go to http://127.0.0.1:8033/clouds
- Check index topic redirects to homepage - http://127.0.0.1:8033/t/juju-documentation/1087, http://127.0.0.1:8033/juju-documentation/1087, http://127.0.0.1:8033/t/1087, http://127.0.0.1:8033/1087 all go to http://127.0.0.1:8033/
- Check all the URLs in the left navigation are "pretty" (as in `/{something}` rather than `/t/{something}/{id}`)
- Check 404s still work
